crates/net/discv4/src/lib.rs

```diff
@@ -514,7 +514,7 @@ impl Discv4Service {
         let key = kad_key(node_id);
```

Comment:
Interesting - what do you think is the implication there? Is it a bug?

Reply:
> Interesting - what do you think is the implication there? Is it a bug?

yes tech is a bug because we use the callback that this is supposed to hit to remove peers from our internal peer table thus causing a small memory leak

Reply:
@mattsse let's fix in follow-up

Reply:
![image](https://user-images.githubusercontent.com/44074786/207406950-e84a78f3-d768-41a0-ac8d-33829e4b29fe.png)